src/lib.rs

```diff
@@ -1090,8 +1409,8 @@ impl Error for ParseRatioError {
 }
 
 impl RatioErrorKind {
-    fn description(&self) -> &'static str {
-        match *self {
+    fn description(self) -> &'static str {
+        match self {
```

Comment:
Why did you make this change?  It's not really a problem either way, but it's totally unrelated to the rest of this PR.

Reply:
I believe clippy caught this, and since it was minor I thought I'd go ahead and fix. I think there were a few more things that clippy suggested, but I held off on those since they were bigger changes. I'll keep these things out in the future.

Reply:
I figured... clippy means well, but this code is on the error path, unlikely to ever make any difference to performance.  No worries though, the change is harmless.

Reply:
Ahh, yeah good point.